salt/metalk8s/orchestrate/upgrade/node.sls

```diff
@@ -0,0 +1,61 @@
+{%- set dest_version = pillar.orchestrate.dest_version %}
+{%- set node_name = pillar.orchestrate.node_name %}
+
+{%- set kubeconfig = "/etc/kubernetes/admin.conf" %}
+{%- set context = "kubernetes-admin@kubernetes" %}
+
+{%- if pillar.metalk8s.nodes[node_name]['version'] == dest_version %}
+
+Node {{ node_name }} already in version {{ dest_version }}:
+  test.succeed_without_changes
+
+{%- else %}
+
+  {%- set control_plane_ip = salt.saltutil.runner(
+      'mine.get',
+      tgt=pillar.bootstrap_id,
+      fun='control_plane_ip'
+  )[pillar.bootstrap_id] %}
+
+Cordon node {{ node_name }}:
+  metalk8s_kubernetes.node_cordoned:
+    - name: {{ node_name }}
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }}
+
+# TODO: Call drain
+
+Run the highstate on {{ node_name }}:
```

Comment:
I think we should *first* set the version label, *then* cordon/drain/apply highstate/uncordon, and stop.

Reply:
In which case the pillar override wouldn't be required.

Reply:
Hmm for me we should set the version label only when the node is really running this version so at the end. 
And then we not execute anything on node which are already in the correct version and if we got an error during drain cordon highstate or uncordon we don't change the label version.

Reply:
No. We're intent-driven, Kubernetes-style. We set the 'desired' version, then we apply actions (as generic as possible) to make that 'desire' reality.

Reply:
Could we have a short session today to discuss this and the path forward, @TeddyAndrieux @gdemonet ?

Reply:
Ok following the discussion we had, lets
- have one orchestrate for all the cluster upgrade
- set the version before upgrading a node
- always do cordon, drain, uncordon

Reply:
This PR is only about one node, I will do the global orchestrate in another PR